pkg/kv/kvserver/replica_lifecycle_datadriven_test.go

```diff
@@ -318,6 +339,80 @@ func TestReplicaLifecycleDataDriven(t *testing.T) {
 				rs.replica = nil
 				return output
 
+			case "append-raft-entries":
+				rangeID := dd.ScanArg[roachpb.RangeID](t, d, "range-id")
+				numEntries := dd.ScanArg[int](t, d, "num-entries")
+				rs := tc.mustGetRangeState(t, rangeID)
+				require.NotNil(t, rs.replica, "replica must be created before appending entries")
+
+				batch := tc.storage.NewBatch()
+				defer batch.Close()
+
+				sl := logstore.NewStateLoader(rangeID)
+				lastIndex := rs.replica.ts.Index
+				rs.replica.ts.Index += kvpb.RaftIndex(numEntries)
+				term := rs.replica.hs.Term
+
+				for i := 0; i < numEntries; i++ {
+					entryIndex := lastIndex + 1 + kvpb.RaftIndex(i)
+					require.NoError(t, storage.MVCCBlindPutProto(
+						ctx, batch,
+						sl.RaftLogKey(entryIndex), hlc.Timestamp{},
+						&raftpb.Entry{Index: uint64(entryIndex), Term: uint64(term)},
+						storage.MVCCWriteOptions{},
+					))
+				}
+
+				output, err := print.DecodeWriteBatch(batch.Repr())
+				require.NoError(t, err)
+				err = batch.Commit(true)
+				require.NoError(t, err)
+				return strings.ReplaceAll(output, "\n\n", "\n")
+
+			case "create-range-data":
+				rangeID := dd.ScanArg[roachpb.RangeID](t, d, "range-id")
+				numUserKeys := dd.ScanArgOr[int](t, d, "num-user-keys", 0)
+				numSystemKeys := dd.ScanArgOr[int](t, d, "num-system-keys", 0)
+				numLockTableKeys := dd.ScanArgOr[int](t, d, "num-lock-table-keys", 0)
+				require.True(t, numUserKeys > 0 || numSystemKeys > 0 || numLockTableKeys > 0)
+
+				rs := tc.mustGetRangeState(t, rangeID)
+				batch := tc.storage.NewBatch()
+				defer batch.Close()
+
+				ts := hlc.Timestamp{WallTime: 1}
+
+				// 1. User keys.
+				for i := 0; i < numUserKeys; i++ {
+					key := append(rs.desc.StartKey.AsRawKey(), byte(i))
```

Comment:
Could be worthy or not: consider making these appended bytes start from `'a'`, or also have some delimiter. Worth doing if this makes the printed keys in the dd output more readable.

Rn (which isn't bad now that I look at it):
```
Put: 0.000000001,0 /Local/Range"c\x00"/Transaction/...
Put: 0.000000001,0 /Local/Range"c\x01"/Transaction/...
```

Could be (not sure if it's how it would be printed):
```
Put: 0.000000001,0 /Local/Range/c-a/Transaction/...
Put: 0.000000001,0 /Local/Range/c-b/Transaction/...
```